frontend/src/components/applyForm/utils.tsx

```diff
@@ -0,0 +1,197 @@
+import { RJSFSchema } from "@rjsf/utils";
+import { ErrorObject } from "ajv";
+import { get as getSchemaObjectFromPointer } from "json-pointer";
+import { filter, get } from "lodash";
+
+import { JSX } from "react";
+
+import { UiSchema } from "./types";
+import { FieldsetWidget } from "./widgets/FieldsetWidget";
+import SelectWidget from "./widgets/SelectWidget";
+import TextareaWidget from "./widgets/TextAreaWidget";
+import TextWidget from "./widgets/TextWidget";
+
+export function buildFormTreeClosure(
```

Comment:
[incredibly inconsequential nit] I feel like `buildForTreeRecursive` or `buildFormTreeNested` is a better term - the `buildFormTree` defined in here is closing over the accumulator array, but I don't think we need to call out the closure, especially since later on I'd expect this to be implemented as a regular reducer